crates/anvil-polkadot/src/substrate_node/service.rs

```diff
@@ -0,0 +1,154 @@
+use polkadot_sdk::{
+    sc_basic_authorship, sc_consensus, sc_consensus_manual_seal,
+    sc_executor::WasmExecutor,
+    sc_network,
+    sc_service::{self, error::Error as ServiceError, Configuration, RpcHandlers, TaskManager},
+    sc_transaction_pool, sp_io,
+    sp_runtime::traits::Block as BlockT,
+    sp_timestamp,
+    substrate_frame_rpc_system::SystemApiServer,
+};
+use std::sync::Arc;
+use substrate_runtime::{OpaqueBlock as Block, RuntimeApi};
+
+use crate::AnvilNodeConfig;
+
+pub type FullClient =
+    sc_service::TFullClient<Block, RuntimeApi, WasmExecutor<sp_io::SubstrateHostFunctions>>;
+
+pub type Backend = sc_service::TFullBackend<Block>;
+
+pub type TransactionPoolHandle = sc_transaction_pool::TransactionPoolHandle<Block, FullClient>;
+
+type SelectChain = sc_consensus::LongestChain<Backend, Block>;
+
+pub struct Service {
+    pub task_manager: TaskManager,
+    pub client: Arc<FullClient>,
+    pub backend: Arc<Backend>,
+    pub tx_pool: Arc<TransactionPoolHandle>,
+    pub rpc_handlers: RpcHandlers,
+}
+
+/// Builds a new service for a full client.
+pub fn new<Network: sc_network::NetworkBackend<Block, <Block as BlockT>::Hash>>(
+    _anvil_config: &AnvilNodeConfig,
+    config: Configuration,
+) -> Result<Service, ServiceError> {
+    let (client, backend, keystore_container, mut task_manager) =
+        sc_service::new_full_parts::<Block, RuntimeApi, _>(
+            &config,
+            None,
+            sc_service::new_wasm_executor(&config.executor),
+        )?;
+    let client = Arc::new(client);
+
+    let transaction_pool = Arc::from(
+        sc_transaction_pool::Builder::new(
+            task_manager.spawn_essential_handle(),
+            client.clone(),
+            config.role.is_authority().into(),
+        )
+        .with_options(config.transaction_pool.clone())
+        .build(),
+    );
+
+    let import_queue = sc_consensus_manual_seal::import_queue(
+        Box::new(client.clone()),
+        &task_manager.spawn_essential_handle(),
+        None,
+    );
+
+    let net_config = sc_network::config::FullNetworkConfiguration::<
+        Block,
+        <Block as BlockT>::Hash,
+        Network,
+    >::new(&config.network, None);
+    let metrics = Network::register_notification_metrics(None);
+
+    let (network, system_rpc_tx, tx_handler_controller, sync_service) =
+        sc_service::build_network(sc_service::BuildNetworkParams {
```

Comment:
I think we can try to strip out as much as possible. We should not need the networking stack. If you remove the network, you would also need to remove the call to `spawn_tasks` below. Internally some tasks from there are still useful, like to keep the transaction pool updated. So we should spawn some of these tasks manually here.

Reply:
good idea, done